File: algorithms/442.cpp
#include <iostream>
#include <vector>
using namespace std;

class Solution {
public:
    vector<int> findDuplicates(vector<int> &nums) {
        vector<int> result;
        for (auto num: nums) {
            num = abs(num);
            if (nums[num-1] < 0) {
                result.push_back(num);
            } else {
                nums[num-1] *= -1;
            }
        }
        return result;
    }
};

int main() {
    return 0;
}

File: algorithms/python3/442.py
class Solution:
    def findDuplicates(self, nums):
        """
        :type nums: List[int]
        :rtype: List[int]
        """
        for i in range(len(nums)):
            while nums[i] != i+1 and nums[i] != nums[nums[i]-1]:
                nums[nums[i]-1], nums[i] = nums[i], nums[nums[i]-1]
        r = []
        for i in range(len(nums)):
            if nums[i] != i+1:
                r.append(nums[i])
        return r
